src/App.tsx

```diff
@@ -3,58 +1,141 @@
-      <div className="tile is-ancestor">
-        <div className="tile is-parent">
-          <div className="tile is-child box is-success">
-            <div className="block">
-              <UserSelector />
-            </div>
+export const App = () => {
+  const [users, setUsers] = useState<User[] | null>(null);
+  const [selectedUser, setSelectedUser] = useState<User | null>(null);
+  const [selectedPost, setSelectedPost] = useState<Post | null>(null);
+  const [isOpenedNewCommentForm, setIsOpenedNewCommentForm] = useState(false);
+  const [posts, setPosts] = useState<Post[] | null>(null);
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState('');
+  const showNoPost =
+    selectedUser && !isLoading && error.length === 0 && posts?.length === 0;
+  const showPostList =
+    selectedUser &&
+    !isLoading &&
+    error.length === 0 &&
+    posts &&
+    posts.length > 0;
 
-            <div className="block" data-cy="MainContent">
-              <p data-cy="NoSelectedUser">No user selected</p>
+  useEffect(() => {
+    getUsers()
+      .then(setUsers)
+      .catch(usersLoadingError => {
+        setError('usersLoadingError');
 
-              <Loader />
+        throw usersLoadingError;
+      });
+  }, []);
 
-              <div
-                className="notification is-danger"
-                data-cy="PostsLoadingError"
-              >
-                Something went wrong!
-              </div>
+  useEffect(() => {
+    if (selectedUser) {
+      setIsLoading(true);
+      // setError(false);
+      setError('');
+      setPosts(null);
+      setSelectedPost(null);
+      setIsOpenedNewCommentForm(false);
+
+      getPosts(selectedUser.id)
+        .then(setPosts)
+        .catch(loadingError => {
+          // setError(true);
+          setError('postsLoadingError');
+
+          throw loadingError;
+        })
+        .finally(() => {
+          setIsLoading(false);
+        });
+    }
+  }, [selectedUser]);
 
-              <div className="notification is-warning" data-cy="NoPostsYet">
-                No posts yet
+  return (
+    <main className="section">
+      <div className="container">
+        <div className="tile is-ancestor">
+          <div className="tile is-parent">
+            <div className="tile is-child box is-success">
+              <div className="block">
+                {users && users.length > 0 && (
+                  <UserSelector
+                    usersList={users}
+                    selectedUser={selectedUser}
+                    onSelectUser={setSelectedUser}
+                  />
+                )}
               </div>
 
-              <PostsList />
+              <div className="block" data-cy="MainContent">
+                {users && !selectedUser && (
+                  <p data-cy="NoSelectedUser">No user selected</p>
+                )}
+
+                {isLoading && <Loader />}
+
+                {!isLoading && error.length > 0 && (
```

Comment:
The error handling here displays a generic message 'Something went wrong!'. Consider displaying the actual error message or a more specific message to help users understand what went wrong.